core/shared/src/test/scala/fs2/ChunkSpec.scala

```diff
@@ -1,128 +1,21 @@
-//     "uncons.long" in propUncons[Long, Chunk[Long]]
-//     "uncons.unspecialized" in propUncons[Int, Chunk[Int]]
-//
-//     "isempty.boolean" in propIsEmpty[Boolean, Chunk[Boolean]]
-//     "isempty.byte" in propIsEmpty[Byte, Chunk[Byte]]
-//     "isempty.double" in propIsEmpty[Double, Chunk[Double]]
-//     "isempty.long" in propIsEmpty[Long, Chunk[Long]]
-//     "isempty.unspecialized" in propIsEmpty[Int, Chunk[Int]]
-//
-//     "filter.boolean" in propFilter[Boolean, Chunk[Boolean]]
-//     "filter.byte" in propFilter[Byte, Chunk[Byte]]
-//     "filter.double" in propFilter[Double, Chunk[Double]]
-//     "filter.long" in propFilter[Long, Chunk[Long]]
-//     "filter.unspecialized" in propFilter[Int, Chunk[Int]]
-//
-//     "foldleft.boolean" in propFoldLeft[Boolean, Chunk[Boolean]]
-//     "foldleft.byte" in propFoldLeft[Byte, Chunk[Byte]]
-//     "foldleft.double" in propFoldLeft[Double, Chunk[Double]]
-//     "foldleft.long" in propFoldLeft[Long, Chunk[Long]]
-//     "foldleft.unspecialized" in propFoldLeft[Int, Chunk[Int]]
-//
-//     "foldright.boolean" in propFoldRight[Boolean, Chunk[Boolean]]
-//     "foldright.byte" in propFoldRight[Byte, Chunk[Byte]]
-//     "foldright.double" in propFoldRight[Double, Chunk[Double]]
-//     "foldright.long" in propFoldRight[Long, Chunk[Long]]
-//     "foldright.unspecialized" in propFoldRight[Int, Chunk[Int]]
-//
-//     "toarray.boolean" in propToArray[Boolean, Chunk[Boolean]]
-//     "toarray.byte" in propToArray[Byte, Chunk[Byte]]
-//     "toarray.double" in propToArray[Double, Chunk[Double]]
-//     "toarray.long" in propToArray[Long, Chunk[Long]]
-//     "toarray.unspecialized" in propToArray[Int, Chunk[Int]]
-//
-//     "concat.boolean" in propConcat[Boolean, Chunk[Boolean]]
-//     "concat.byte" in propConcat[Byte, Chunk[Byte]]
-//     "concat.double" in propConcat[Double, Chunk[Double]]
-//     "concat.long" in propConcat[Long, Chunk[Long]]
-//     "concat.unspecialized" in propConcat[Int, Chunk[Int]]
-//
-//     "map.boolean => byte" in forAll { c: Chunk[Boolean] =>
-//       (c map { b => if (b) 0.toByte else 1.toByte }).toArray shouldBe (c.toArray map { b => if (b) 0.toByte else 1.toByte })
-//     }
-//
-//     "map.long => long" in forAll { c: Chunk[Long] => (c map (1 +)).toArray shouldBe (c.toArray map (1 +)) }
-//   }
-// }
+package fs2
+
+import cats.kernel.laws.discipline.EqTests
+import cats.laws.discipline.{ MonadTests, TraverseTests }
+import cats.implicits._
+import org.scalacheck.{ Arbitrary, Gen }
+
+class ChunkSpec extends Fs2Spec {
+  implicit val arbByteChunk: Arbitrary[Chunk[Byte]] = Arbitrary {
+    for {
+      n <- Gen.choose(0, 100)
+      values <- Gen.containerOfN[Array, Byte](n, Arbitrary.arbByte.arbitrary)
+      offset <- Gen.choose(0, n)
+      sz <- Gen.choose(0, n - offset)
+    } yield Chunk.bytes(values, offset, sz)
+  }
+
+  checkAll("Eq[Chunk[Byte]]", EqTests[Chunk[Byte]].eqv)
+  checkAll("Monad[Chunk.Bytes]", MonadTests[Chunk].monad[Byte, Byte, Byte])
```

Comment:
This is a bit awkward: `Chunk.Bytes` is not the right kind to be a `Monad`, but we need to test all the specializations, as that's where the bugs are.  If we proceed with this, we'll have Chunk.Ints, Chunk.Longs, etc.

Maybe we should pass the `arbByteChunk` instance in explicitly?

Reply:
Ah yeah, no complaints here either way